tests/functional/platforms/08-warn-if-regex-matches-localhost.t

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env bash
+# THIS FILE IS PART OF THE CYLC WORKFLOW ENGINE.
+# Copyright (C) NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#-------------------------------------------------------------------------------
+# Check that ``[platforms][localhost]`` is only set automatically if it
+# not set in ``global.cylc``.
+
+export REQUIRE_PLATFORM=''
+. "$(dirname "$0")/test_header"
+
+set_test_number 4
+
+create_test_global_config "" "
+    [platforms]
+        [[localh...]]
+        # This shouldn't override `localh...` in this one case, because the
+        # localhost default pins it to the top of the list.
+        [[localhost]]
+            [[[meta]]]
+                foo = 'foo'
+"
```

Comment:
Quoting error which should cause the test to fail:

```suggestion
create_test_global_config '' '
    [platforms]
        [[localh...]]
        # This shouldn't override `localh...` in this one case, because the
        # localhost default pins it to the top of the list.
        [[localhost]]
            [[[meta]]]
                foo = 'foo'
'
```

Reply:
(unresolving)